test/box-tap/CMakeLists.txt

```diff
@@ -1 +1,72 @@
 build_module(check_merge_source check_merge_source.c)
+
+set(TEST_SUITE_NAME "box-tap")
+
+message(STATUS "Add test suite ${TEST_SUITE_NAME}")
+
+tarantool_make_lua_path(LUA_PATH
+  PATHS
+    ${PROJECT_SOURCE_DIR}/test-run/?.lua
+)
+
+# XXX: The call produces both test and target <box-tap-deps>
+# as a side effect.
+_add_test_suite_target(${TEST_SUITE_NAME}
+  LABELS ${TEST_SUITE_NAME}
+  DEPENDS check_merge_source
+)
+
+list(APPEND TEST_ENV
+  "BUILDDIR=${PROJECT_BINARY_DIR}"
+  "LUA_PATH=${LUA_PATH}"
+)
+
+file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR} *.test.lua)
+foreach(test_path ${tests})
+  get_filename_component(test_name ${test_path} NAME)
+  # FIXME: By default, GLOB lists directories.
+  # Directories are omitted in the result if LIST_DIRECTORIES
+  # is set to false. New in version CMake 3.3.
+  if(${test_name} STREQUAL ${TEST_SUITE_NAME})
+    continue()
+  endif()
+  set(test_title "test/${TEST_SUITE_NAME}/${test_name}")
+  set(work_dir "${CMAKE_CURRENT_BINARY_DIR}/${test_name}-workdir")
+  file(MAKE_DIRECTORY ${work_dir})
+  set(TEST_CMD "${RM} -rf ${work_dir}/* && ${TARANTOOL_BIN} ${test_path}")
+  add_test(NAME ${test_title}
+           COMMAND ${BASH} -c "${TEST_CMD}"
+           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+  # `LISTEN` is required by the following tests:
+  # - test/box-tap/auth.test.lua
+  # - test/box-tap/extended_error.test.lua
+  set(TEST_ENV_VARS "${TEST_ENV};LISTEN=${work_dir}/listen.sock;")
+  if(NOT ${test_title} STREQUAL test/box-tap/cfg.test.lua AND
```

Comment:
Q: Why do we avoid the `TT_WORK_DIR` for these tests?
Please add the comment.

Reply:
Added:

```diff
--- a/test/box-tap/CMakeLists.txt
+++ b/test/box-tap/CMakeLists.txt
@@ -42,6 +42,8 @@ foreach(test_path ${tests})
   # - test/box-tap/auth.test.lua
   # - test/box-tap/extended_error.test.lua
   set(TEST_ENV_VARS "${TEST_ENV};LISTEN=${work_dir}/listen.sock;")
+  # The tests below could be broken by setting environment
+  # variable `TT_WORK_DIR`, so `TT_WORK_DIR` is not set.
   if(NOT ${test_title} STREQUAL test/box-tap/cfg.test.lua AND
      NOT ${test_title} STREQUAL test/box-tap/gh-4562-errno-at-xdir_scan.test.lua)
     set(TEST_ENV_VARS ${TEST_ENV_VARS};TT_WORK_DIR=${work_dir};)
```